management_interface/management_interface/models.py

```diff
@@ -1,56 +1,92 @@
 import hashlib
+import uuid
 
+from django.contrib.auth.models import User
+from django.core.validators import EmailValidator
 from django.db import models
-import uuid
-from datetime import datetime
+from django.utils import timezone
+
+
+class SecureEmailValidator(EmailValidator):
+    message = "Enter an nhs.net email address"
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs, allowlist=["nhs.net", "thepsc.co.uk"])
+
+    def validate_domain_part(self, _):
+        return False
 
 
 class BaseModel(models.Model):
     """
     Base model holding shared data
     """
-    id = models.CharField(
-        primary_key=True, max_length=64, default=uuid.uuid4, editable=False, unique=True
-    )
-    created_at = models.DateTimeField(default=datetime.utcnow, editable=False)
+
+    class Meta:
+        abstract = True
+
+    id = models.CharField(primary_key=True, max_length=64, default=uuid.uuid4, editable=False, unique=True)
+    created_at = models.DateTimeField(default=timezone.now, editable=False)
     updated_at = models.DateTimeField(auto_now=True, editable=False)
+    created_by = models.ForeignKey(
+        User, on_delete=models.CASCADE, null=True, editable=False, related_name="%(app_label)s_%(class)s_created_by"
```

Comment:
`models.CASCADE` can't be right, can it?  We don't want deleting a user to delete any of the data, or vice versa.

If this is being difficult to sort out, I'd suggest that we put `created_by` and `updated_by` to one side and we can raise a separate PR.  It's not critical behaviour so we shouldn't let it hold us up.